scripts/build.signer.sh

```diff
@@ -24,11 +24,20 @@ CANDID_FILE="$(jq -r .canisters.signer.candid dfx.json)"
 WASM_FILE="$(jq -r .canisters.signer.wasm dfx.json)"
 ARG_FILE="$(jq -r .canisters.signer.init_arg_file dfx.json)"
 BUILD_DIR="target/wasm32-unknown-unknown/release"
+COMMIT_FILE="target/commit"
 
 ####
 # Builds the Wasm without metadata
 cargo build --locked --target wasm32-unknown-unknown --release -p signer
 
+####
+# Gets commit and tag information, if available.
```

Comment:
If we run `dfx build signer`, we can query git to get this information.  If we are running inside docker or some other constrained environment, `.git` is unavailable but `target/commit` is copied in.  If we are in some other constrained environment, we just ensure that the metadata file exists, even if it is empty, so that we don't break builds.